### PR TITLE
[RFC] Restore compatibility with doctrine/dbal 2.9

### PR DESCRIPTION
**Pull request by ausi**
Followup to #238.

### CODE REVIEW

File: core-bundle/src/Doctrine/Schema/DcaSchemaProvider.php [hunk 1]
```diff
@@ -316,7 +313,20 @@ private function parseIndexSql(Table $table, $keyName, $sql)
                 $flags[] = 'fulltext';
             }
 
-            $table->addIndex($columns, $matches[2], $flags);
+            $options = [];
+
+            if (array_sum($lengths)) {
```
Comment (leofeyer) [inline core-bundle/src/Doctrine/Schema/DcaSchemaProvider.php]: This is nice. 👍 Although I really wonder why null values do not trigger a warning with `declare(strict_types=1)`.

File: core-bundle/src/Doctrine/Schema/DcaSchemaProvider.php [hunk 2]
```diff
@@ -316,7 +313,20 @@ private function parseIndexSql(Table $table, $keyName, $sql)
...
+                $options['lengths'] = $lengths;
+
+                // Backwards compatibility for doctrine/dbal < 2.9
+                if (!method_exists(AbstractPlatform::class, 'supportsColumnLengthIndexes')) {
+                    $columns = array_combine($columns, array_map(function($column, $length) {
+                        return $column . '(' . $length . ')';
```
Comment (leofeyer) [inline core-bundle/src/Doctrine/Schema/DcaSchemaProvider.php]: Shouldn't we check `null !== $length` here?

Comment (ausi) [reply core-bundle/src/Doctrine/Schema/DcaSchemaProvider.php]: Done in c714532aa25394b60cd75790bffc6e559be56d7b

File: core-bundle/src/Doctrine/Schema/DcaSchemaProvider.php (hunk 1 above)
Comment (ausi) [reply core-bundle/src/Doctrine/Schema/DcaSchemaProvider.php]: We could also use

```suggestion
            if (array_filter($lengths)) {
```

instead. Should we?

Comment (leofeyer) [reply core-bundle/src/Doctrine/Schema/DcaSchemaProvider.php]: Yes.

Comment (ausi) [reply core-bundle/src/Doctrine/Schema/DcaSchemaProvider.php]: Done in edb3c933828520f7c7bac39f4d2e060fa184950f

**Review by leofeyer — APPROVED** [review]

File: core-bundle/tests/Doctrine/Schema/DcaSchemaProviderTest.php [hunk 3]
```diff
@@ -323,7 +324,57 @@ public function testHandlesIndexesWithKeyLength()
...
+            $this->assertSame('string', $table->getColumn('col'.$i)->getType()->getName());
+            $this->assertSame(255, $table->getColumn('col'.$i)->getLength());
+        }
+
+        $this->assertTrue($table->hasIndex('col123'));
+        $this->assertFalse($table->getIndex('col123')->isUnique());
+        $this->assertSame([100, null, 99], $table->getIndex('col123')->getOption('lengths'));
+        if (method_exists(AbstractPlatform::class, 'supportsColumnLengthIndexes')) {
+            $this->assertSame(['col1', 'col2', 'col3'], $table->getIndex('col123')->getColumns());
+        }
```
Comment (Toflar) [inline core-bundle/tests/Doctrine/Schema/DcaSchemaProviderTest.php]: CS

**Review by Toflar — APPROVED** [review]
Minor CS remark.

File: core-bundle/tests/Doctrine/Schema/DcaSchemaProviderTest.php (hunk 3 above)
Comment (ausi) [reply core-bundle/tests/Doctrine/Schema/DcaSchemaProviderTest.php]: Done in e755e203f997ef9d950eb61e35a700a242e9225a

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**